### PR TITLE
Migrate from woocommerce.com to woo.com

### LINKED ISSUE #10175: Support woo.com universal links in the app

### PR DESCRIPTION
**Pull request by kidinov**
Closes: #10175

### Description

The PR migrates the usage of the woocommerce.com URLs to woo.com
* Universal links: both remains should be supported
* Links that lead from the app to the browser were replaced with the links that currently redirected to anyway
* Repository documentation

### Testing instructions

* Valudate that universal links work, both with `woocommerce.com` and `woo.com` domains. E.g.:
```
 adb shell am start -a android.intent.action.VIEW \
   -c android.intent.category.BROWSABLE \
   -d "https://woocommerce.com/mobile/payments/tap-to-pay"
```
image

* Validate that all links replacements lead to the same page as before

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by kidinov** [comment]
@0nko could you please take a look if it's safe/needed to change the URL 
[here](https://github.com/woocommerce/woocommerce-android/blob/6dbc7fa4d975a5b2b67ba3bcd716176fbc22476f/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/plans/PlansViewModel.kt#L66)
and 
[here](https://github.com/woocommerce/woocommerce-android/blob/6dbc7fa4d975a5b2b67ba3bcd716176fbc22476f/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/domain/DomainPurchaseViewModel.kt#L33)?

There is a redirect, so I am even not sure that this works right now

@hafizrahman the same question for [this part](https://github.com/woocommerce/woocommerce-android/blob/6dbc7fa4d975a5b2b67ba3bcd716176fbc22476f/WooCommerce/src/main/kotlin/com/woocommerce/android/extensions/WCSSRModelExt.kt#L250). That looks like just a text so should be fine, but just in case

**Comment by hafizrahman** [comment]
> @hafizrahman the same question for [this part](https://github.com/woocommerce/woocommerce-android/blob/6dbc7fa4d975a5b2b67ba3bcd716176fbc22476f/WooCommerce/src/main/kotlin/com/woocommerce/android/extensions/WCSSRModelExt.kt#L250). That looks like just a text so should be fine, but just in case

@kidinov I saw that in Core they replaced that string with "Connected to Woo.com" ([reference](https://github.com/woocommerce/woocommerce/blob/0a98a249b027b16cbf10490202db6798e9b1179a/plugins/woocommerce/includes/admin/views/html-admin-page-status-report.php#L740-L742))

So it would be good if you can add the change to the string you linked to as well. Thanks for noticing!

File: WooCommerce/src/main/kotlin/com/woocommerce/android/AppUrls.kt [hunk 1]
```diff
@@ -51,45 +51,45 @@ object AppUrls {
...
-    const val WISEPAD_3_MANUAL_CARD_READER = "https://woocommerce.com/wp-content/uploads/2022/12/wp3_product_sheet.pdf"
+        "https://woo.com/wp-content/uploads/2022/12/c2xbt_product_sheet.pdf"
+    const val M2_MANUAL_CARD_READER = "https://woo.com/wp-content/uploads/2022/12/m2_product_sheet.pdf"
+    const val WISEPAD_3_MANUAL_CARD_READER = "https://woo.com/wp-content/uploads/2022/12/wp3_product_sheet.pdf"
 
     const val PLAY_STORE_APP_PREFIX = "http://play.google.com/store/apps/details?id="
 
     const val LOGIN_WITH_EMAIL_WHAT_IS_WORDPRESS_COM_ACCOUNT =
-        "https://woocommerce.com/document/what-is-a-wordpress-com-account/"
+        "https://woo.com/woocommerce-features/"
```
Comment (hafizrahman) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/AppUrls.kt]: The page there does not seem to explain what a WordPress.com account is.

Did you want to use this instead?

https://woo.com/document/what-is-a-wordpress-com-account/

**Review by hafizrahman — DISMISSED** [review]
I left a couple of comments re: updating SSR and about the link explaining WordPress.com account.

The rest of the link changes look good to me, I checked them manually and they do open to the right thing.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/AppUrls.kt (hunk 1 above)
Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/AppUrls.kt]: Thanks! Good catch. Fixed here - 4ec1db284309254440f2c05d014596b679d42dbe

**Comment by kidinov** [comment]
@hafizrahman 

>@kidinov I saw that in Core they replaced that string with "Connected to Woo.com" ([reference](https://github.com/woocommerce/woocommerce/blob/0a98a249b027b16cbf10490202db6798e9b1179a/plugins/woocommerce/includes/admin/views/html-admin-page-status-report.php#L740-L742))

> So it would be good if you can add the change to the string you linked to as well. Thanks for noticing!

Thanks for confirming that. Fixed here
a4dd6aab0ebafb704eb1851622e5a4b01e2eeebc

**Comment by hafizrahman** [comment]
Looks good on my side. For the question for Ondrej, he's AFK today so pinging @JorgeMucientes to check if the "woocommerce.com" as exit check logic is safe to be changed:

> please take a look if it's safe/needed to change the URL
[here](https://github.com/woocommerce/woocommerce-android/blob/6dbc7fa4d975a5b2b67ba3bcd716176fbc22476f/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/plans/PlansViewModel.kt#L66)
and
[here](https://github.com/woocommerce/woocommerce-android/blob/6dbc7fa4d975a5b2b67ba3bcd716176fbc22476f/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/domain/DomainPurchaseViewModel.kt#L33)?

**Comment by 0nko** [comment]
> There is a redirect, so I am even not sure that this works right now

These are not links, but triggers in the WebView that notify the ViewModel. I've tested the plan and domain purchase and it seems like both domains work, which is weird 🤔. Can you double check if you see the same behaviour with "woo.com"?

**Comment by kidinov** [comment]
@0nko As I have 0 knowledge about the flows, and even not sure where this code runs, would you mind changing the URL to `https://woo.com/` and see if it's ok?

**Comment by 0nko** [comment]
These are the Woo express plan purchase and domain purchase flows. I've checked them both and exiting the purchase WebView works. 👍

**Comment by kidinov** [comment]
Thank you @0nko 🙏

I've changed the trigger URL then - df765f30f201d2e7d500f4bb62b3b5e057be315e

**Review by JorgeMucientes — APPROVED** [review]

**Comment by JorgeMucientes** [comment]
Good work @kidinov ! Thanks for handling this migration. I've tested the WooCommerce domain deeplinks as well as smoked tested many of the url updated in the PR. Everything works as expected. Ready for merging 👍🏼

**Pull request merged**